### PR TITLE
Move unshipped API to shipped

### PR DESCRIPTION
**Pull request by Evangelink**
## Description

As discussed during our sync, we want to always make API changes directly to the shipped file. This should not prevent us to consider doing a "breaking" change while we are only in preview.

Fixes [AB#1526141](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1526141)

### CODE REVIEW

File: src/Microsoft.TestPlatform.CoreUtilities/PublicAPI/PublicAPI.Shipped.txt
Comment (Evangelink) [inline src/Microsoft.TestPlatform.CoreUtilities/PublicAPI/PublicAPI.Shipped.txt:168]: Could we make this `internal` with some `InternalsVisibileTo` instead?

File: src/Microsoft.TestPlatform.CrossPlatEngine/PublicAPI/PublicAPI.Shipped.txt
Comment (Evangelink) [inline src/Microsoft.TestPlatform.CrossPlatEngine/PublicAPI/PublicAPI.Shipped.txt:128]: We are mainly using `AbortXXX` in the codebase, any reason not to go with similar naming here?

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/PublicAPI/PublicAPI.Shipped.txt
Comment (Evangelink) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/PublicAPI/PublicAPI.Shipped.txt:205]: Any reason to make these 2 APIs `public` instead of `internal`? We probably can't change the others but if that's not too late for this one I would vote for adding a new resource file with accessibility set to `internal` to avoid adding more surface to support.

File: src/Microsoft.TestPlatform.CrossPlatEngine/PublicAPI/PublicAPI.Shipped.txt
Comment (nohwnd) [reply src/Microsoft.TestPlatform.CrossPlatEngine/PublicAPI/PublicAPI.Shipped.txt:128]: @cvpoienaru ^^^

Comment (Evangelink) [reply src/Microsoft.TestPlatform.CrossPlatEngine/PublicAPI/PublicAPI.Shipped.txt:128]: Just pointing out that we released 17.2 so if we expect anyone to have started using this API (I don't think anyone outside us and Test Explorer would) we shouldn't change.

File: .editorconfig
Comment (Evangelink) [inline .editorconfig:47]: Maybe that would be better to assume all files have indent of 2 and make specific case indent 4.

**Review by cvpoienaru — APPROVED** [review]

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx
Comment (Evangelink) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: This thread that got closed: https://github.com/microsoft/vstest/pull/3584#discussion_r853257612

File: src/Microsoft.TestPlatform.CoreUtilities/PublicAPI/PublicAPI.Shipped.txt
Comment (Evangelink) [reply src/Microsoft.TestPlatform.CoreUtilities/PublicAPI/PublicAPI.Shipped.txt:168]: @nohwnd Shall I proceed with merge or it this something we want to address.

Comment (nohwnd) [reply src/Microsoft.TestPlatform.CoreUtilities/PublicAPI/PublicAPI.Shipped.txt:168]: go agead

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx
Comment (nohwnd) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: Are you sure they will still get localized that way?

**Comment by Evangelink** [comment]
@nohwnd I need your approval. Thank you.

**Comment by nohwnd** [comment]
https://github.com/microsoft/vstest/pull/3584#discussion_r855926940 

^^^ Waiting for this :)

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx
Comment (Evangelink) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: Do we have a way to ensure that? I can say that the `build.cmd` does touch these files but I don't know how the rest is done.

Comment (Haplois) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: It should get picked by localization team, and we should get soon after this PR is merged. If not we can ping them ask for extra steps.

Comment (Haplois) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: You might need to update: https://github.com/microsoft/vstest/blob/main/Localize/LocProject.json

Comment (Evangelink) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Resources/InternalResources.resx:120]: Done.

**Comment by Evangelink** [comment]
@nohwnd I have replied to your comment and updated the file suggested by @Haplois .

**Review by nohwnd — APPROVED** [review]

**Pull request merged**
